internal/certgen/certgen.go

```diff
@@ -317,6 +412,72 @@ func newCA(r io.Reader, cn string, expiry time.Time) ([]byte, []byte, error) {
 	return keyPEMData, certPEMData, nil
 }
 
+func newTLSSecret(secretname, namespace string, keyPEM, certPEM []byte) *corev1.Secret {
+
+	return &corev1.Secret{
+		Type: corev1.SecretTypeTLS,
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Secret",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secretname,
+			Namespace: namespace,
+			Labels: map[string]string{
+				"app": "contour",
+			},
+		},
+		Data: map[string][]byte{
+			corev1.TLSCertKey:       certPEM,
+			corev1.TLSPrivateKeyKey: keyPEM,
+		},
+	}
+}
+
+func newCertOnlySecret(secretname, namespace, certfilename string, certPEM []byte) *corev1.Secret {
+
+	return &corev1.Secret{
+		Type: corev1.SecretTypeOpaque,
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Secret",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secretname,
+			Namespace: namespace,
+			Labels: map[string]string{
+				"app": "contour",
+			},
+		},
+		Data: map[string][]byte{
+			path.Base(certfilename): certPEM,
+		},
+	}
+}
+
+func newClient(kubeconfig string, inCluster bool) (*kubernetes.Clientset, error) {
```

Comment:
i'm not sure this function belongs in this package. What the other functions that rely on newClient need is the *kubernetes.ClientSet. It feels super weird that internal/certgen needs to know if its being called from a process which is running inside or outside a cluster.

Probably the way to solve this is to move newClient's caller, Gencerts into cmd/contour then it can reuse the existing NewClient function there